audio-worklet/src/noisegate-demo.js

```diff
@@ -0,0 +1,284 @@
+    this.noiseGain_ = new GainNode(this.context_);
+    this.speechAndNoiseGain_ = new GainNode(this.context_);
+    this.withoutNoisegateGain_ = new GainNode(this.context_, {gain: 0});
+    this.noisegateGain_ = new GainNode(this.context_);
+    this.scriptProcessorGain_ = new GainNode(this.context_);
+
+    this.noiseGain_.connect(this.speechAndNoiseGain_);
+    this.speechGain_.connect(this.speechAndNoiseGain_);
+    this.speechAndNoiseGain_.connect(this.withoutNoisegateGain_)
+        .connect(this.masterGain_);
+    
+    this.speechAndNoiseGain_.connect(this.noisegateScriptProcessor_.input);
+    this.noisegateScriptProcessor_.output.connect(this.scriptProcessorGain_)
+        .connect(this.noisegateGain_).connect(this.masterGain_);
+
+    this.masterGain_.connect(this.context_.destination);
+
+    this.loadSound('sound/speech.ogg').then((sound) => {
+      this.speechBuffer_ = sound;
+      this.sourceButton_.enable();
+    });
+
+    this.loadSound('sound/white_noise.ogg').then((sound) => {
+      this.noiseBuffer_ = sound;
+    });
+
+    this.playing_ = false;
+  }
+
+  async loadSound(url) {
+    const response = await fetch(url);
+    const sound = await response.arrayBuffer();
+    return this.context_.decodeAudioData(sound);
+  }
+
+  /**
+   * Initalize HTML elements when document has loaded.
+   * @param {String} containerId Id of parent container.
+   * @param {String} workletButtonId Id of worklet radio button.
+   * @param {String} scriptProcessorButtonId Id of scriptProcessor radio button.
+   * @param {String} noisegateId Id of noisegate radio button.
+   * @param {String} withoutNoisegateId Id of withoutNoisegate radio button.
+   */
+  initializeGUI(containerId, workletButtonId, scriptProcessorButtonId,
+                noisegateId, withoutNoisegateId) {
+    this.sourceButton_ = new SourceController(
+        containerId, this.start.bind(this), this.stop.bind(this));
+
+    this.thresholdSlider_ =
+        new ParamController(containerId, this.setThreshold.bind(this), {
+          type: 'range',
+          min: this.thresholdMin_,
+          max: this.thresholdMax_,
+          step: 1,
+          default: this.threshold_,
+          name: 'Threshold (dBFS)'
+        });
+    this.thresholdSlider_.disable();
+
+    this.attackSlider_ =
+        new ParamController(containerId, this.setAttack.bind(this), {
+          type: 'range',
+          min: this.attackMin_,
+          max: this.attackMax_,
+          step: 0.01,
+          default: this.attack_,
+          name: 'Attack (s)'
+        });
+    this.attackSlider_.disable();
+
+    this.releaseSlider_ =
+        new ParamController(containerId, this.setRelease.bind(this), {
+          type: 'range',
+          min: this.releaseMin_,
+          max: this.releaseMax_,
+          step: 0.01,
+          default: this.release_,
+          name: 'Release (s)'
+        });
+    this.releaseSlider_.disable();
+
+    this.gainSlider_ =
```

Comment:
`gainSlider` => `masterGainSlider`?  You have so many gain nodes it's hard to tell what this is.